native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -184,13 +195,13 @@ private static Object convertElement(BXmlItem xmlItem, String attributePrefix,
             }
             children = convertToArray(fieldType, children);
         }
-        return insertDataToMap(childrenData, children, rootNode,  keyValue, fieldType);
+        return new Object[]{fieldName, insertDataToMap(childrenData, children, rootNode,  fieldName, fieldType)};
```

Comment:
why we change this

Reply:
When a field has the name annotation, both names (xml element and record field) are different. So, we have to return the field name to the next process. Otherwise, we have to process its name there again. I have updated the impl in c53ab77. Please review that.